File: app/src/main/java/com/tematikhonov/androidgeekbrainssecondapp/data/CardsSource.java
package com.tematikhonov.androidgeekbrainssecondapp.data;

public interface CardsSource {
    CardsSource init(CardsSourceResponce cardsSourceResponce);
    Note getNote(int position);
    int size();
    void deleteCardData(int position);
    void updateCardData(int position, Note note);
    void clearCardData();
    void addCardData(Note note);
}

File: app/src/main/java/com/tematikhonov/androidgeekbrainssecondapp/data/CardsSource.kt
package com.tematikhonov.androidgeekbrainssecondapp.data

interface CardsSource {
    fun init(cardsSourceResponce: CardsSourceResponce?): CardsSource?
    fun getNote(position: Int): Note?
    fun size(): Int
    fun deleteCardData(position: Int)
    fun updateCardData(position: Int, note: Note?)
    fun clearCardData()
    fun addCardData(note: Note?)
}